### PR TITLE
New CmdLet to get UsageAggregates

### PR DESCRIPTION
**Pull request by Matt-Westphal**
This command will get the usage data for a subscription.

Also added a script to verify the module works as expected

The test is marked as skipped since I can get the test framework to start after working at it for 4 days.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Matt-Westphal__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/ResourceManager/Commerce/Commands.UsageAggregates/TestTheCmdLet.ps1 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿$subscription = <Add subscription ID>
+
+Import-Module C:\GitHub\MyGitHub\azure-powershell\src\Package\Debug\ResourceManager\AzureResourceManager\UsageAggregates\Microsoft.Azure.Commands.UsageAggregates.dll
```
Comment (markcowl) [inline src/ResourceManager/Commerce/Commands.UsageAggregates/TestTheCmdLet.ps1]: Please do not check in files with hardcoded paths

File: src/ResourceManager/Resources/Commands.Resources/AzureResourceManager.psd1 [hunk 2]
```diff
@@ -98,7 +98,8 @@ NestedModules = @(
     '.\ResourceManager\Microsoft.Azure.Commands.ResourceManager.Cmdlets.dll',
     '.\ApiManagement\Microsoft.Azure.Commands.ApiManagement.dll',
     '.\StorageManagement\Microsoft.Azure.Commands.Management.Storage.dll',
-    '.\OperationalInsights\Microsoft.Azure.Commands.OperationalInsights.dll'
+    '.\OperationalInsights\Microsoft.Azure.Commands.OperationalInsights.dll',
+	'.\UsageAggregates\Microsoft.Azure.Commands.UsageAggregates.dll'
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/AzureResourceManager.psd1]: fix spacing

**Comment by markcowl** [comment]
This PR should also include:  (1) Updated installer that includes your files (use the BuildInstaller script in the Tools folder); (2) dll help file; (3) Changes to changelog.txt describing your new cmdlet

File: src/ResourceManager/Commerce/Commands.UsageAggregates/GetUsageAggregatesCommand.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    using Commerce.UsageAggregates.Models;
+    using WindowsAzure.Commands.Utilities.Common;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.Get, "UsageAggregates"), OutputType(typeof(UsageAggregationGetResponse))]
+    public class GetUsageAggregatesCommand : AzurePSCmdlet
+    {
+        [Parameter(Mandatory = true, HelpMessage = "The start of the time range to retrieve data for.")]
+        public DateTime ReportedStartTime { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/Commerce/Commands.UsageAggregates/GetUsageAggregatesCommand.cs]: Why ReportedStartTiem and ReportedEndTime and not just StartTime and EndTime?

File: src/ResourceManager/Commerce/Commands.UsageAggregates/GetUsageAggregatesCommand.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        public DateTime ReportedEndTime { get; set; }
+
+        private AggregationGranularity aggregationGranularity = AggregationGranularity.Daily;
+        [Parameter(Mandatory = false, HelpMessage = "Value is either daily (default) or hourly to tell the API how to return the results grouped by day or hour.")]
+        public AggregationGranularity AggregationGranularity { 
+            get { return aggregationGranularity; } 
+            set { aggregationGranularity = value;}
+        }
+
+        private bool showDetails = true;
```
Comment (markcowl) [inline src/ResourceManager/Commerce/Commands.UsageAggregates/GetUsageAggregatesCommand.cs]: Please use _ for private fields, and include them in the section above the public properties

File: src/ResourceManager/Commerce/Commands.UsageAggregates.Test/ScenarioTests/UsageAggregatesTests.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+using Xunit;
+
+namespace Microsoft.Azure.Commands.UsageAggregates.Test.ScenarioTests
+{
+
+    public class UsageAggregatesTests 
+    {
+      
+         [Fact]
```
Comment (markcowl) [inline src/ResourceManager/Commerce/Commands.UsageAggregates.Test/ScenarioTests/UsageAggregatesTests.cs]: <nit> remove space </nit>

**Comment by markcowl** [comment]
@azuresdkci retest this please

File: src/ResourceManager/Commerce/Commands.UsageAggregates/GetUsageAggregatesCommand.cs (hunk 3 above)
Comment (Matt-Westphal) [inline src/ResourceManager/Commerce/Commands.UsageAggregates/GetUsageAggregatesCommand.cs]: With usage data there is multiple uses of DateTime.  Reported vs when the usage was consumed.  which is a critical distinction to make sure they consumers have all the data they need.  Also this is consistent with output of the billing data from the Azure system

**Comment by Matt-Westphal** [comment]
Where can I find some documentation or any idea where/how to do these?  Would have included them if I had known they needed to be done.

**Comment by hovsepm** [comment]
please pull from upstream and resolve conflicts.

**Comment by pmsousa** [comment]
Hi,
I think I might have found a bug. When calling the cmdlet Get-UsageAggregates I get a error message:
```
Get-UsageAggregates -ReportedStartTime "2015-08-20" -ReportedEndTime "2015-08-21" -Debug

Get-UsageAggregates : The request is invalid.
At line:1 char:1
+ Get-UsageAggregates -ReportedStartTime "2015-08-20" -ReportedEndTime  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-UsageAggregates], CloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.UsageAggregates.GetUsageAggregatesCommand

```

I thought the problem could have been with the date format,so I've tried several different ways: dd/mm/yyyy, yyyy/mm/dd, etc. Always the same result.

Finally, I've used the -debug flag and these were my findings:

```
DEBUG: [Common.Authentication]: Checking token expiration, token expires '09/11/2015 16:00:27 +00:00' Comparing to '09/11/2015 15:04:10 +00:
00' With threshold '00:05:00', calculated time until token expiry: '00:56:16.8425779'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/subscriptions/{SUBSCRIPTIONID}/providers/Microsoft.Commerce/UsageAggregates?api-version=201
5-06-01-preview&reportedstartTime=20%2F08%2F2015 00:00:00&reportedEndTime=21%2F08%2F2015 00:00:00,&showDetails=true&aggregationGranularity=D
aily

Headers:

Body:



DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Pragma                        : no-cache
x-ms-ratelimit-remaining-subscription-reads: 14998
x-ms-request-id               : e5e62bf5-d2a3-4604-aa59-23652c33613c
x-ms-correlation-request-id   : e5e62bf5-d2a3-4604-aa59-23652c33613c
x-ms-routing-request-id       : NORTHEUROPE:20150911T150411Z:e5e62bf5-d2a3-4604-aa59-23652c33613c
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Cache-Control                 : no-cache
Date                          : Fri, 11 Sep 2015 15:04:11 GMT
Server                        : Microsoft-IIS/8.0
X-AspNet-Version              : 4.0.30319
X-Powered-By                  : ASP.NET

Body:
{
  "message": "The request is invalid."
}

```

If you note the URI, the reportedstartTime and reportedEndTime fields have spaces between the date and time; I think theses should be URL Encoded or set to ShortDate format!



PS: I've replaced my subscription ID with the text {SUBSCRIPTIONID}

**Pull request merged**